Clarify What a PostStart Hook Does (#46731)
kubernetes/website: content/en/docs/concepts/containers/container-lifecycle-hooks.md

the Kubernetes management system executes the handler according to the hook action,
`httpGet` , `tcpSocket` and `sleep` are executed by the kubelet process, and `exec` is executed in the container.

Hook handler calls are synchronous within the context of the Pod containing the Container.
This means that for a `PostStart` hook,

> the Container ENTRYPOINT and hook fire asynchronously.

Author: The word "asynchronously" is wrong here.
If you start a pod and set the `PostStart` to sleep for 180 seconds, it finishes its execution before the ENTRYPOINT begins. This is not the same for the `PreStop` hook due to the TERM signal. Try the below examples and let me know what you think.

**post-start.yaml**
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: post-start
spec:
  containers:
  - name: alpine-container
    image: alpine:latest
    lifecycle:
      postStart:
        exec:
          command: ["/bin/sh", "-c", "echo 'Container has started' > /tmp/poststart.log && sleep 180"]
    command: ["/bin/sh", "-c", "while true; do sleep 30; done"]
```

**pre-stop.yaml**
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pre-stop
spec:
  containers:
  - name: alpine-container
    image: alpine:latest
    lifecycle:
      preStop:
        exec:
          command: ["/bin/sh", "-c", "echo 'Container is stopping' > /tmp/prestop.log && sleep 180"]
    command: ["/bin/sh", "-c", "while true; do sleep 30; done"]
```

Reviewer: ok makes sense

> Hook handler calls like `PostStart` are executed when a container is created, 

Reviewer 2: ```suggestion
Hook handler calls like `PostStart`, are initiated when a container is created,
```

> Hook handler calls like `PostStart`, are initiated when a container is created,

Reviewer 3: ```suggestion
Hook handler calls such as `PostStart`, are initiated when a container is created,
```

Author: It makes sense because I am introducing only one example `PodStart`, and not comparing it with others. Thanks!

> Hook handler calls such as `PostStart`, are initiated when a container is created,

Author: You're right, @matthyx. Here is the fix, let me know what you think?
```suggestion
The `PostStart` hook handler call is initiated when a container is created,
```

> meaning the container ENTRYPOINT and the `PostStart` hook start at the same time. 

Reviewer 2: ```suggestion
meaning the container ENTRYPOINT and the `PostStart` hook are triggered simultaneously. 
```

> However, if the `PostStart` hook takes too long to run or hangs, 

Reviewer 2: ```suggestion
However, if the `PostStart` hook takes too long to execute or if it hangs,
```

> it can prevent the container from reaching a `running` state.

Reviewer 2: ```suggestion
it can prevent the container from transitioning to a `running` state.
```
